huaweicloud/services/evs/resource_huaweicloud_evs_snapshot_group.go

```diff
@@ -0,0 +1,413 @@
+		"instant_access":        utils.ValueIgnoreEmpty(d.Get("instant_access")),
+		"name":                  utils.ValueIgnoreEmpty(d.Get("name")),
+		"description":           utils.ValueIgnoreEmpty(d.Get("description")),
+		"enterprise_project_id": utils.ValueIgnoreEmpty(epsID),
+		"tags":                  utils.ValueIgnoreEmpty(d.Get("tags")),
+	}
+	if d.Get("instant_access").(bool) {
+		body["instant_access"] = true
+	}
+	return map[string]interface{}{"snapshot_group": utils.RemoveNil(body)}
+}
+
+func buildUpdateSnapshotGroupBodyParams(d *schema.ResourceData) map[string]interface{} {
+	body := map[string]interface{}{
+		"name":        d.Get("name"),
+		"description": d.Get("description"),
+	}
+	return map[string]interface{}{"snapshot_group": body}
+}
+
+func GetSnapshotGroupDetail(client *golangsdk.ServiceClient, snapshotGroupID string) (interface{}, error) {
+	requestPath := client.Endpoint + "v5/{project_id}/snapshot-groups/{snapshot_group_id}"
+	requestPath = strings.ReplaceAll(requestPath, "{project_id}", client.ProjectID)
+	requestPath = strings.ReplaceAll(requestPath, "{snapshot_group_id}", snapshotGroupID)
+	requestOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+	}
+
+	resp, err := client.Request("GET", requestPath, &requestOpt)
+	if err != nil {
+		return nil, err
+	}
+
+	return utils.FlattenResponse(resp)
+}
+
+func resourceEvsSnapshotGroupCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	var (
+		cfg     = meta.(*config.Config)
+		region  = cfg.GetRegion(d)
+		httpUrl = "v5/{project_id}/snapshot-groups"
+		product = "evs"
+		epsID   = cfg.GetEnterpriseProjectID(d)
+	)
+	client, err := cfg.NewServiceClient(product, region)
+	if err != nil {
+		return diag.Errorf("error creating EVS client: %s", err)
+	}
+
+	requestPath := client.Endpoint + httpUrl
+	requestPath = strings.ReplaceAll(requestPath, "{project_id}", client.ProjectID)
+	requestOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		JSONBody:         utils.RemoveNil(buildCreateSnapshotGroupBodyParams(d, epsID)),
+	}
+
+	retryFunc := func() (interface{}, bool, error) {
+		resp, err := client.Request("POST", requestPath, &requestOpt)
+		retry, err := handleMultiOperationsError(err)
+		return resp, retry, err
+	}
+
+	r, err := common.RetryContextWithWaitForState(&common.RetryContextWithWaitForStateParam{
+		Ctx:          ctx,
+		RetryFunc:    retryFunc,
+		Timeout:      d.Timeout(schema.TimeoutCreate),
+		DelayTimeout: 1 * time.Second,
+		PollInterval: 10 * time.Second,
+	})
+
+	if err != nil {
+		return diag.Errorf("error creating EVS snapshot group: %s", err)
+	}
+
+	respBody, err := utils.FlattenResponse(r.(*http.Response))
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	snapshotGroupId := utils.PathSearch("snapshot_group_id", respBody, "").(string)
+	if snapshotGroupId == "" {
+		return diag.Errorf("error creating EVS snapshot group: ID is not found in API response")
+	}
+
+	d.SetId(snapshotGroupId)
+
```

Comment:
if jobID := utils.PathSearch("job_id", respBody, "").(string); jobID != "" {
  if err := waitingForSnapshotGroupJobSuccess(ctx, client, jobID, d.Timeout(schema.TimeoutCreate)); err != nil {
    return diag.Errorf("error waiting for EVS snapshot group (%s) job success: %s", d.Id(). err)
  }
}

Refer function `waitingForVolumeJobSuccess` in evs volume resource to write `waitingForSnapshotGroupJobSuccess`.